src/core/interpreter.js

```diff
@@ -87,7 +87,7 @@ const Interpreter = async (
                 data.vars || {},
                 data.object || {},
                 ["roles", "users", "everyone"],
-                data.array || [],
```

Comment:
You're changing into object, this doesn't make sense.

Reply:
It is since the data inside d.array will be arrays.